lib/src/environment/state/environment_state.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:flutter/widgets.dart';
+
+import 'basic_environment_state.dart';
+
+/// Values used to render a [Scene].
+/// TODO improve docs
+class EnvironmentState extends ChangeNotifier {
+  BasicEnvironmentState _basicState = BasicEnvironmentState.defaultState;
+
+  /// TODO
+  BasicEnvironmentState get basicState => _basicState;
```

Comment:
I agree that if you go with this approach, you may as well just use keys and values the same as for custom environment controls. E.g. whatever API we offer to 3P developers should be comfortable enough for us to use for built-in controls too.

Reply:
Done.